gobblin-metrics-libs/gobblin-metrics-base/src/main/java/org/apache/gobblin/metrics/event/EventSubmitter.java

```diff
@@ -79,23 +79,49 @@ public EventSubmitter build() {
     }
   }
 
+  public EventSubmitter(MetricContext context) {
+    this.metricContext = Optional.of(context);
+    this.namespace = GobblinEventBuilder.NAMESPACE;
+    this.metadata = Maps.newHashMap();
+  }
+
+  public void submit(GobblinEventBuilder eventBuilder) {
+    eventBuilder.addAdditionalMetadata(this.metadata);
+    this.metricContext.get().submitEvent(eventBuilder.build());
+  }
+
+  public static void submit(MetricContext context, GobblinEventBuilder builder) {
+    context.submitEvent(builder.build());
+  }
+
+  /**
+   *
+   * @deprecated use {@link #EventSubmitter(MetricContext)}
+   */
+  @Deprecated
```

Comment:
Why is the builder pattern deprecated? How would a user add metadata to the event submitter?

Reply:
We are trying to get away from having metadata in the EventSubmitter. We can use gobblinEventBuilder.addAdditionalMetadata

Reply:
We deprecated it because we no longer want to have namespace in the EventSubmitter and rather have it in the GobblinEventBuilder
Will provide a way to add metadata to the EventSubmitter through the constructor.

Reply:
Why don't you want the namespace in the `EventSubmitter`? Even then, why not just deprecate the namespaced constructor and leave the rest as si?